README.md
# Dart Material SVG Icons
### Optimized for Dart + SASS

[Material Design Icons][materialicons] are the official open-source icons featured in the Google Material Design specification.
This project provides this icon-set for Dart.
 
## What's cool
Googles original version comes "only" with black or white SVGs.  
**material_icons** allows you to **theme** your icons!!!  
Tested with Chrome, Safari, FF + IE.

![Screenshot][screenshot]

## Usage
* Clone the ["purple"][purple] example
  - Your main.scss should look something like this:
  ```css
  $icon-color: nth($palette-primary, 2);
  @import "packages/material_icons/sass/svg-utils";
  @import "material_icons";
  ``` 
* Open <a href="https://rawgit.com/MikeMitterer/dart-material-icons/master/lib/sass/index.html" target="_blank">[this]</a> page
  - Use Chrome or Safari for this!
* Click on icon that you want to include in your project
* Copy the **_material-icons.scss** section
* Paste it into your _material-icons.scss
```css
@import 'packages/material_icons/sass/action/svg/production/ic_account_box_18px';

.bg-ic_account_box_18px {
    @include svg-background("18");
    
    /* you can exclude or delete this line */
    @include svg-fallback("action/svg/production/ic_account_box_18px.svg","action/2x_web/ic_account_box_black_18dp.png","action/2x_web/ic_account_box_white_18dp.png","action/2x_web/ic_account_box_grey600_18dp.png");
    
    background-image: svg-ic_account_box_18px($icon-color);
}
```
* Copy the **index.html** section
* Paste it into your index.html
```html
    <div class="bg-ic_favorite_24px svg-size-24 svg-bg"></div>
```
* In main.scss define your icon-color: `$icon-color : #9B30FF;`
* In web/styles: **sass main.scss main.css -r ../packages/material_icons/sassext/urlencode.rb**
* and: **autoprefixer main.css**

**Note:**
You may ask what the hack???? does that mean:  
-r ../packages/material_icons/sassext/urlencode.rb  
You are adding a custom ruby-function to sass - here it is urlencode(...).  
The inline SVG get urlencoded and makes IE happy.  

That's it.    
Your page should look like <a href="https://rawgit.com/MikeMitterer/dart-material-icons/master/example/purple/web/index.html" target="_blank">[this]</a> page   

###Play with your index.html###

```html
<body class="no-svg fallback-white">
    <div class="bg-ic_favorite_24px svg-size-24 svg-bg"></div>
    <div class="bg-ic_account_balance_24px svg-size-24 svg-bg fallback-grey"></div>
    <div class="bg-ic_accessibility_48px svg-size-48 svg-bg"></div>
</body>
```
**no-svg** - turns off svg and uses png's instead  
**fallback-white** | **fallback-black** | **fallback-grey** - PNG color to use  
This option can also be specified per div (line 3).    

You can [see](http://wsk.mikemitterer.at/) a [reallife] sample using this technique on [wsk.mikemitterer.at](http://wsk.mikemitterer.at/) 

## Features and bugs

Please file feature requests and bugs at the [issue tracker][tracker].

### License

    Copyright 2016 Michael Mitterer (office@mikemitterer.at),
    IT-Consulting and Development Limited, Austrian Branch

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
    either express or implied. See the License for the specific language
    governing permissions and limitations under the License.
    
If this plugin is helpful for you - please [(Circle)](http://gplus.mikemitterer.at/) me  
or **star** this repo here on GitHub.
      
[tracker]: https://github.com/MikeMitterer/dart-material-icons/issues
[live]: https://rawgit.com/MikeMitterer/dart-material-icons/master/lib/sass/index.html
[materialicons]: https://github.com/google/material-design-icons
[purple]: https://github.com/MikeMitterer/dart-material-icons/tree/master/example/purple
[screenshot]: https://github.com/MikeMitterer/dart-material-icons/raw/master/lib/sass/screenshot.png?raw=true
[reallife]: https://github.com/MikeMitterer/dart-wsk-material/tree/master/example


CHANGELOG.md
# Change Log for material_icons
Google Material icons - optimized for Dart + SASS

## [v1.0.4](http://github.com/mikemitterer/dart-material-icons/compare/v1.0.3...v1.0.4) - 2016-11-16

### Docs
* Strong-Mode support [35d7228](https://github.com/mikemitterer/dart-material-icons/commit/35d7228428cb7882c0c86e81dec98de2fd28b4e3)

## [v1.0.3](http://github.com/mikemitterer/dart-material-icons/compare/v1.0.2...v1.0.3) - 2015-12-04

### Docs
* Grinder added [0d710db](https://github.com/mikemitterer/dart-material-icons/commit/0d710dba680d7091199d63a621655c5de967e93e)

## [v1.0.2](http://github.com/mikemitterer/dart-material-icons/compare/v1.0.1...v1.0.2) - 2015-03-20

### Docs
* Readme updated [eb9035a](https://github.com/mikemitterer/dart-material-icons/commit/eb9035a54b090b28cb64edb5e1bbc72a3ee758b7)
* Readme updated [9331ba9](https://github.com/mikemitterer/dart-material-icons/commit/9331ba9933351e86ef41cf309e578798082742c4)

## [v1.0.1](http://github.com/mikemitterer/dart-material-icons/compare/v1.0.0...v1.0.1) - 2015-03-18

### Feature
* Inline SVGs are urlencoded now [0eb8479](https://github.com/mikemitterer/dart-material-icons/commit/0eb84795ffbaa1c9fec217e98c6069e57017b60a)


This CHANGELOG.md was generated with [**Changelog for Dart**](https://pub.dartlang.org/packages/changelog)


test/all_test.dart
// Copyright (c) 2015, <your name>. All rights reserved. Use of this source code
// is governed by a BSD-style license that can be found in the LICENSE file.

library material_icons.test;

import 'package:unittest/unittest.dart';
import 'package:material_icons/material_icons.dart';

main() {
  group('A group of tests', () {
    Awesome awesome;

    setUp(() {
      awesome = new Awesome();
    });

    test('First Test', () {
      expect(awesome.isAwesome, isTrue);
    });
  });
}


tool/grind.dart
import 'package:grinder/grinder.dart';

main(final List<String> args) => grind(args);

@Task()
@Depends(analyze )
build() async {
    // Update Sample
    await runAsync("/Users/mikemitterer/.pub-cache/bin/buildSamples",arguments: [ "-u" ]);

    // Analyze
    analyze();

    // Build!
    await runAsync("/Users/mikemitterer/.pub-cache/bin/buildSamples",arguments: [ "-bc" ]);
}

@Task()
analyze() {
    final List<String> libs = [
        "lib/material_icons.dart"
    ];

    libs.forEach((final String lib) => Analyzer.analyze(lib));

    final List<String> samples = [
        "purple/web/main.dart"
    ];
    samples.forEach((final String sample ) {
        final String sampleFolder = sample.replaceAll("/web/main.dart","");
        run("tool/analyze-sample.sh",arguments: [ "samples/${sampleFolder}", "web/main.dart" ]);
    });

}

@Task()
clean() => defaultClean();


tools/gensvg.sh
#!/bin/bash
##################################################################
#
# Erstellt weiße SVG-Files
#
##################################################################

APPNAME="gensvg.sh"

#------------------------------------------------------------------------------
# BASIS

#LOGFILE="${APPNAME}_`date +"%Y%m%d"`.log"


#------------------------------------------------------------------------------
# Lock wird gesetzt
LOCKDIR="/var/lock"
LOCKFILE="$LOCKDIR/$APPNAME.lock"

[ -f ${LOCKFILE} ] && exit 0
trap 'cleanup' EXIT TERM INT

cleanup () {
    rm -f ${LOCKFILE};
    exit 255;
    }
    
touch ${LOCKFILE}

#------------------------------------------------------------------------------
# Read settings from settings-file
#
SETTINGS_FILE=$APPNAME.conf
SETTINGS_USER="~/.$SETTINGS_FILE"
SETTINGS_ETC="/etc/$SETTINGS_FILE"

if test -e $SETTINGS_USER
then
    . $SETTINGS_USER
elif test -e $SETTINGS_ETC
then
    . $SETTINGS_ETC
fi

DEBUG=false
DEBUG=true

#------------------------------------------------------------------------------
# Functions
#

sampleFunction() {
	PARAM1=$1

	if test "$PARAM1" != ""
	then
	    echo "Param1: $PARAM1"
	else
	    echo "Param1 not set..."
	fi

	for SUBFOLDER in "${SASSFILES[@]}"
	do
		echo -e "\tSubfolder: ${SUBFOLDER}"
	done
}  

_cleanup() {
    ASSETSPATH="../lib/assets"

    DIRSTOREMOVE=(
    "*/svg/design"
    "svg/design"
    "*/drawable-xxxhdpi"
    "*/drawable-xxhdpi"
    "*/drawable-xhdpi"
    "*/drawable-mdpi"
    "*/drawable-hdpi"
    "*/3x_ios"
    "*/2x_web"
    "*/2x_ios"
    "*/1x_web"
    "*/1x_ios"
    )

    for DIR in ${DIRSTOREMOVE[@]}; do
        log "Remove ${ASSETSPATH}/${DIR}..."
        rm -rf "${ASSETSPATH}/${DIR}"
    done
    echo "Unnecessary files remove!"
}

generate() {
    COLORNAME="white"
    COLORCODE="#ffffff"

    _cleanup

    FILES=`find . -iname "*px.svg"`
    for FILE in ${FILES}; do
        FILEWITHOUTEXT=${FILE%*.*}
        COLORFILE="${FILEWITHOUTEXT}-${COLORNAME}.svg"

        if [ -f "${COLORFILE}" ];then
            rm -f ${COLORFILE}
            echo "${COLORFILE} deleted.."
        fi

        echo "${FILEWITHOUTEXT}-${COLORNAME}.svg created!"
        sed "s/<path d=/<path fill=\"${COLORCODE}\" d=/" ${FILEWITHOUTEXT}.svg > ${FILEWITHOUTEXT}-${COLORNAME}.svg;
    done

    echo "Now you can generate your css/scss with './gencss.sh --icons'"
}

gensass() {
    ASSETSPATH="../assets"
    SRCPATH="${ASSETSPATH}/action"
    SRCPATH="${ASSETSPATH}"
    TARGETPATH="../lib/sass"

    SVGICONS="svg-icons.scss"
    SVGUTILS="svg-utils.scss"
    SVGSAMPLES="svg-sample.scss"
    INDEXHTML="index.html"

    _cleanup
    rm -f "${TARGETPATH}/index.html" "${TARGETPATH}/main.css" "${TARGETPATH}/svg-icons.scss" "${TARGETPATH}/svg-sample.scss"
    #rm -f ${MAINSASS}
    #rm -f ${SAMPLESASS}
    #rm -f ${SAMPLEHTML}

    TMPFILE=".temp-svg.tmp"

    # SAMPLE - BASISFARBE
    echo "@import '${SVGICONS}';" >> "${TARGETPATH}/${SVGSAMPLES}"
    echo "@import '${SVGUTILS}';" >> "${TARGETPATH}/${SVGSAMPLES}"
    echo -e "\$icon-color: #666666 !default;\n" >> "${TARGETPATH}/${SVGSAMPLES}"
    #echo -e "\$icon-alpha: 1 !default;\n" >> "${ASSETSPATH}/${SAMPLESASS}"
    #echo -e "\$icon-color-internal: rgba(red(\$icon-color),green(\$icon-color),blue(\$icon-color),\$icon-alpha) !default;\n" >> "${ASSETSPATH}/${SAMPLESASS}"

    prefixhtml "${TARGETPATH}/${INDEXHTML}"

    FILES=`find ${SRCPATH} -iname "*px.svg"`
    for FILE in $FILES; do
        FILE="`echo ${FILE} | sed s#${TARGETPATH}/##g`"

        FILEWITHOUTEXT=${FILE%*.*}
        FILENAME=`basename ${FILE}`
        FILEPATH="`echo ${FILE} | sed s/${FILENAME}//`"
        PUREFILENAME=`basename ${FILE%*.*}`

        SASSFILE=${PUREFILENAME}.scss
        SASSPATH="`echo ${FILEPATH} | sed s#^\.\./assets/##`"
        SASSPATH="${TARGETPATH}/${SASSPATH}";
        SASSTOIMPORT="`echo ${FILEWITHOUTEXT} | sed s#^\.\./assets/##`"

        PNGFILE="`echo ${FILEWITHOUTEXT} | sed s#/svg/#/2x_web/#g`"
        PNGFILE="`echo ${PNGFILE} | sed s/_18px//g`"
        PNGFILE="`echo ${PNGFILE} | sed s/_24px//g`"
        PNGFILE="`echo ${PNGFILE} | sed s/_36px//g`"
        PNGFILE="`echo ${PNGFILE} | sed s/_48px//g`"
        PNGFILE="`echo ${PNGFILE} | sed s#production/##g`"
        PNGFILE="`echo ${PNGFILE} | sed s#^\./##`"

        if [[ ${PUREFILENAME} =~ .*24px.* ]]
        then
            SIZE="24"
        fi
        if [[ ${PUREFILENAME} =~ .*48px.* ]]
        then
            SIZE="48"
        fi
        if [[ ${PUREFILENAME} =~ .*36px.* ]]
        then
            SIZE="36"
        fi
        if [[ ${PUREFILENAME} =~ .*18px.* ]]
        then
            SIZE="18"
        fi

        FALLBACKBLACK="${PNGFILE}_black_${SIZE}dp.png"
        FALLBACKGREY="${PNGFILE}_grey600_${SIZE}dp.png"
        FALLBACKWHITE="${PNGFILE}_white_${SIZE}dp.png"
        FALLBACKSVG="`echo ${FILE} | sed s#^\./##`"

        FALLBACKBLACK="`echo ${FALLBACKBLACK} | sed s#^\.\./assets/##`"
        FALLBACKGREY="`echo ${FALLBACKGREY} | sed s#^\.\./assets/##`"
        FALLBACKWHITE="`echo ${FALLBACKWHITE} | sed s#^\.\./assets/##`"
        FALLBACKSVG="`echo ${FALLBACKSVG} | sed s#^\.\./assets/##`"

        log "--------------------------------------------------------------"
        log "Filepath:           ${FILEPATH}"
        log "File without Ext:   ${FILEWITHOUTEXT}"
        log "Pure Filename:      ${PUREFILENAME}"
        log "SASSFile:           ${SASSFILE}"
        log "SASSPath:           ${SASSPATH}"
        log "SassToImport:       ${SASSTOIMPORT}"
        log "PNG-File:           ${PNGFILE}"
        log "Size:               ${SIZE}"
        log "SrcPath:            ${SRCPATH}"
        log "File:               ${FILE}"
        log "FallbackBlack       ${FALLBACKBLACK}"
        log "FallbackGrey        ${FALLBACKGREY}"
        log "FallbackWhite       ${FALLBACKWHITE}"
        log "FallbackSVG         ${FALLBACKSVG}"
        log "--------------------------------------------------------------\n"

        if [ -f "../${FALLBACKBLACK}" ];then
            log "FallbackBlack: ${FALLBACKBLACK} available..."
        fi
        if [ -f "../${FALLBACKGREY}" ];then
            log "FallbackGrey: ${FALLBACKGREY} available..."
        fi
        if [ -f "../${FALLBACKWHITE}" ];then
            log "FallbackWhite: ${FALLBACKWHITE} available..."
        fi
        if [ -f "../${FALLBACKSVG}" ];then
            log "FallbackSVG: ${FALLBACKSVG} available..."
        fi
        log "--------------------------------------------------------------\n"

        if [ ! -d "${SASSPATH}" ]; then
            mkdir -p ${SASSPATH}
        fi

        if [ -f "${SASSPATH}/${SASSFILE}" ];then
            rm -f "${SASSPATH}/${SASSFILE}"
            log "${SASSFILE} deleted.."
        fi

        echo "Preparing: ${SASSFILE}, size: ${SIZE}..."
        rm -f ${TMPFILE}

        sed "s/<path d=/<path fill=\"\' + \$fillColor + \'\" d=/" "${FILE}" > ${TMPFILE}

        log "Sassfile: ${SASSFILE} created!"

        echo "@function svg-${PUREFILENAME}(\$fillColor) {" > "${SASSPATH}/${SASSFILE}"
        # BASE64
        #echo -n "    @return url('data:image/svg+xml;charset=utf-8;base64,' + base64Encode('" >> "${SASSPATH}/${SASSFILE}"

        # URLENCODE
        echo -n "    @return url('data:image/svg+xml;charset=utf-8,' + urlencode('" >> "${SASSPATH}/${SASSFILE}"
            tr -d "\n" < ${TMPFILE} >> "${SASSPATH}/${SASSFILE}"
        echo -n "'));" >> "${SASSPATH}/${SASSFILE}"

        echo -e "\n}" >> "${SASSPATH}/${SASSFILE}"

        # SASS mit allen IMPORTS
        echo "@import '${SASSTOIMPORT}';" >> "${TARGETPATH}/${SVGICONS}"

        # SAMPLE - definiert HGs
        echo ".bg-${PUREFILENAME} {" >> "${TARGETPATH}/${SVGSAMPLES}"
        echo "     @include svg-background(\"${SIZE}px\");" >> "${TARGETPATH}/${SVGSAMPLES}"
        #echo "     background-repeat: no-repeat;" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "     background-size: 100% 100%;" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "     background-size: contain;" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "     background-position: 50% 50%;" >> "${ASSETSPATH}/${SAMPLESASS}"
        echo "     @include svg-fallback(\"${FALLBACKSVG}\",\"${FALLBACKBLACK}\",\"${FALLBACKWHITE}\",\"${FALLBACKGREY}\");" >> "${TARGETPATH}/${SVGSAMPLES}"

        #echo "     .no-svg & {" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "         &.fallback-black {" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "             background-image: url(${FALLBACKBLACK});" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo -e "         }\n" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "         &.fallback-white {" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "             background-image: url(${FALLBACKWHITE});" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo -e "         }\n" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "         &.fallback-grey {" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "             background-image: url(${FALLBACKGREY});" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo -e "         }\n" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo -e "     }\n" >> "${ASSETSPATH}/${SAMPLESASS}"
        #
        #echo "      .svg-not-inline & {" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "            background-image: url(${FALLBACKSVG});" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo -e "     }\n" >> "${ASSETSPATH}/${SAMPLESASS}"

        echo "     background-image: svg-${PUREFILENAME}(\$icon-color);" >> "${TARGETPATH}/${SVGSAMPLES}"
        #echo "     background-image: svg-${PUREFILENAME}(rgba(red(\$icon-color),green(\$icon-color),blue(\$icon-color),\$icon-alpha));" >> "${ASSETSPATH}/${SAMPLESASS}"
        echo -e "}\n" >> "${TARGETPATH}/${SVGSAMPLES}"

        #echo "      @-moz-document url-prefix() { .bg-${PUREFILENAME} {" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo "            background-image: url(${FALLBACKSVG});" >> "${ASSETSPATH}/${SAMPLESASS}"
        #echo -e "     }}\n" >> "${ASSETSPATH}/${SAMPLESASS}"


        echo "    <div class=\"svgbg\">" >> "${TARGETPATH}/${INDEXHTML}"
        echo "        <div class=\"bg-${PUREFILENAME} svg-size-${SIZE} svg-bg onclick-menu\" tabindex=\"0\">" >> "${TARGETPATH}/${INDEXHTML}"
        echo "            <div class=\"onclick-menu-content\">"  >> "${TARGETPATH}/${INDEXHTML}"
        echo "              <div class=\"filename\">_material-icons.scss:</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "                <div class=\"import\">@import 'packages/material_icons/sass/${SASSTOIMPORT}';</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "                  <div class=\"cssclass\">.bg-${PUREFILENAME} {</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "                     <div class=\"props\">@include svg-background(\"${SIZE}\");</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "                     <div class=\"props\">@include svg-fallback(\"${FALLBACKSVG}\",\"${FALLBACKBLACK}\",\"${FALLBACKWHITE}\",\"${FALLBACKGREY}\");</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "                     <div class=\"props\">background-image: svg-${PUREFILENAME}(\$icon-color);</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "                  <div class=\"close-cssclass\">}</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "              <br>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "              <div class=\"filename\">index.html:</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "                <div class=\"html\">&lt;div class=&quot;bg-${PUREFILENAME} svg-size-${SIZE} svg-bg&quot;&gt;&lt;/div&gt;</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "            </div>"  >> "${TARGETPATH}/${INDEXHTML}"
        echo "        </div>"  >> "${TARGETPATH}/${INDEXHTML}"
        echo "        <div class=\"name\">${PUREFILENAME}</div>" >> "${TARGETPATH}/${INDEXHTML}"
        echo "    </div>" >> "${TARGETPATH}/${INDEXHTML}"
    done
    postfixhtml "${TARGETPATH}/${INDEXHTML}"
    rm -f ${TMPFILE}

    #sassc "${ASSETSPATH}/main.scss" "${ASSETSPATH}/main.css" && autoprefixer "${ASSETSPATH}/main.css"
    sass "${TARGETPATH}/main.scss" "${TARGETPATH}/main.css" -r "../lib/sassext/urlencode.rb" && autoprefixer "${TARGETPATH}/main.css"

    echo -e "\nmain.css generated, you are ready to go!"

}

prefixhtml() {
    HTMLFILE=$1

    echo "<!doctype html>" >> ${HTMLFILE}
    echo "<html lang="en-US">" >> ${HTMLFILE}
    echo "<head>" >> ${HTMLFILE}
    echo "<title>Material Icons</title>" >> ${HTMLFILE}
    echo "<meta charset=utf-8>" >> ${HTMLFILE}
    echo "<meta http-equiv=X-UA-Compatible content="IE=edge">" >> ${HTMLFILE}
    echo "<meta name=description content="Helper to select the right import for material-icons">" >> ${HTMLFILE}
    echo "<meta name=viewport content="width=device-width, initial-scale=1">" >> ${HTMLFILE}
    echo "<meta name=mobile-web-app-capable content=yes>" >> ${HTMLFILE}
    echo "<meta name=apple-mobile-web-app-capable content=yes>" >> ${HTMLFILE}
    echo "<meta name=apple-mobile-web-app-status-bar-style content=black>" >> ${HTMLFILE}
    echo "<meta name=apple-mobile-web-app-title content="Material Icons">" >> ${HTMLFILE}

    echo "<link rel="stylesheet" type="text/css" href="main.css"/>" >> ${HTMLFILE}

    echo "</head>" >> ${HTMLFILE}
    echo "<body>" >> ${HTMLFILE}
}

postfixhtml() {
    HTMLFILE=$1
    echo "</body>" >> ${HTMLFILE}
    echo "</html>" >> ${HTMLFILE}
}

log() {
    STRINGTOLOG=$1

    if [ "${DEBUG}" = true ] ; then
        echo -e ${STRINGTOLOG}
    fi
}

#------------------------------------------------------------------------------
# Options
#

usage()
    {
    echo
    echo "Usage: `basename $0` [ options ]"
    echo -e "\t--gen    Generate white SVG-Files"
    echo -e "\t--sass   Generate white SASS-Files"
    echo
    }


case "$1" in 
    help|-help|--help)
	    usage
	;;
	
    sample|-sample|--sample)
	    sampleFunction "Hallo"
	;;

    gen|-gen|--gen)
	    generate
	;;

    sass|-sass|--sass)
        gensass
    ;;

*)
    usage
	;;

esac    

#------------------------------------------------------------------------------
# Alles OK...

exit 0